src/rshim_pcie.c

```diff
@@ -554,5 +554,8 @@ int rshim_pcie_init(void)
     rshim_pcie_probe(dev);
   }
 
+  if (!dev)
+	  return -1;
+
```

Comment:
The above loop doesn't return. So it'll always come here with dev equals to NULL. Probably you meant if no pcie device was found in the above loop then return -1?

Reply:
Right, stupid me. I'll push a new version setting a boolean when we find the first device.